src/main/java/com/epam/university/java/core/task001/Task001Impl.java

```diff
@@ -1,134 +1,64 @@
 package com.epam.university.java.core.task001;
 
+import com.epam.university.java.core.utils.Validator;
+
 public class Task001Impl implements Task001 {
-    public static final String firstNumberMessageForIllegalArgumentException = "first number must be the value";
-    public static final String secondNumberMessageForIllegalArgumentException = "second number must be the value";
-    public static final String firstNumberMessageForNumberFormatException = "can't convert first input value to number";
-    public static final String secondNumberMessageForNumberFormatException = "can't convert second input value to number";
+
+    public static final String errorMessageForFirstParameterIfLengthNull =
+            "first number's length can't be null";
+    public static final String errorMessageForSecondParameterIfLengthNull =
+            "first number's length can't be null";
+    public static final String firstNumberMessageForNumberFormatException =
+            "can't convert first input value to number";
+    public static final String secondNumberMessageForNumberFormatException =
+            "can't convert second input value to number";
 
     @Override
     public double addition(String firstNumber, String secondNumber) {
-        if (firstNumber == "") {
-            throw new IllegalArgumentException(firstNumberMessageForIllegalArgumentException);
-        }
-        if (firstNumber == null) {
-            throw new IllegalArgumentException(firstNumberMessageForIllegalArgumentException);
-        }
-        if (secondNumber == "") {
-            throw new IllegalArgumentException(secondNumberMessageForIllegalArgumentException);
-        }
-        if (secondNumber == null) {
-            throw new IllegalArgumentException(secondNumberMessageForIllegalArgumentException);
-        }
-        double firstNumberConverted;
-        double secondNumberConverted;
-        try {
-            firstNumberConverted = Double.parseDouble(firstNumber);
-        } catch (NumberFormatException ex) {
-            throw new NumberFormatException(firstNumberMessageForNumberFormatException);
-        }
-        try {
-            secondNumberConverted = Double.parseDouble(secondNumber);
-        } catch (NumberFormatException ex) {
-            throw new NumberFormatException(secondNumberMessageForNumberFormatException);
-        }
-        return firstNumberConverted + secondNumberConverted;
+        Validator.validateValueAndLengthNotNull(firstNumber, secondNumber,
```

Comment:
It's preferably to store default values of error messages in validator itself.

Reply:
Respected Alexander Barmin!
Thank u for the comments - all fulfiled:
- Task005 - complited; 
- Validator and all it's users have been enhanced 
(new default messages added;
 literals comparing by equals; 
using forms of disjunctive operators where possible.)